File: PyEntelechySystem/Core/Unit/BaseUnit.py
"""
@File   : BaseUnit.py
@Author : Ethan Lin
@Date   : 2022/07/05
@Desc   : 
"""

from PyEntelechySystem import *

from PyEntelechySystem.Core.Object.BaseObject import BaseObject
from PyEntelechySystem.Core.Content.Content import Content

@dataclass
class BaseUnit(BaseObject):
    """
    基件
    """

    type_name: str = 'base unit type'
    # self.uid = uuid.uuid4()
    content: Content = Content(type_name='base unit type', head='base unit content head', body='base unit content body')

    # def __int__(self, *args, **kwargs):
    #     self.type_name: str = 'base unit type'
    #     # self.uid = uuid.uuid4()
    #     self.content: Content = Content(type_name='base unit type', content_head='base unit content head', content_body='base unit content body')
    #     super().__init__()
    #     pass

    pass

File: CppEntelechySystem/Core/Unit/BaseUnit.cpp
//
// Created by Ethan on 2022/06/03.
//

#include "BaseUnit.h"

BaseUnit::BaseUnit(const std::string &unit_type_name,
                   const Content &content,
                   const Content &feedback_sign,
                   const Object &object) :
        type_name(unit_type_name),
        content(content),
        feedback_sign(feedback_sign),
        Object(object) {

}
